app/models/finance/statement/ecf.rb

```diff
@@ -2,7 +2,8 @@
 
 class Finance::Statement::ECF < Finance::Statement
   has_one :lead_provider, through: :cpd_lead_provider
-
+  has_many :participant_declarations,        -> { not_voided.not_ineligible }, foreign_key: :statement_id
+  has_many :voided_participant_declarations, -> { voided }, foreign_key: :statement_id, class_name: "ParticipantDeclaration::ECF"
```

Comment:
```suggestion
  has_many :participant_declarations, foreign_key: :statement_id
  has_many :active_participant_declarations, -> { state: %w[eligible payable paid] }, foreign_key: :statement_id
```

- this way the base association does not have scope attached to it like a default scope. i think developers would be surprised to see a default scope. they'll just work around and implement `all_participant_declarations` otherwise
- whitelisting means the addition of new states will not cause bugs here

call voided as:
```ruby
statement.participant_declarations.voided
```

- i think the statement does not really care about voided and this behaviour can be left on the declaration class